flask_static_digest/cli.py

```diff
@@ -1,21 +1,83 @@
-import click
+from dataclasses import dataclass
+from warnings import warn
 
+import click
 from flask import current_app
 from flask.cli import with_appcontext
 
-from flask_static_digest.digester import compile as _compile
 from flask_static_digest.digester import clean as _clean
+from flask_static_digest.digester import compile as _compile
+
+
+@dataclass
+class CompressionAlgo:
+    gzip: bool = False
+    brotli: bool = False
 
 
 @click.group()
-def digest():
-    """md5 tag and gzip static files."""
-    pass
+@click.pass_context
+@with_appcontext
+def digest(ctx):
+    """md5 tag and compress static files."""
+
+    ctx.obj = comp_algo = CompressionAlgo()
+    flask_static_comp = current_app.config.get(
+        "FLASK_STATIC_COMPRESSION", None
+    )
+    flask_static_digest_gzip = current_app.config.get(
+        "FLASK_STATIC_DIGEST_GZIP_FILES", None
+    )
+    if flask_static_comp is not None and flask_static_digest_gzip is not None:
```

Comment:
I'm not sure we need this condition?

- A user sets `FLASK_STATIC_COMPRESSION` and everything is good to go
- A user sets `FLASK_STATIC_DIGEST_GZIP_FILES` and they receive a deprecation warning

In both cases, the user is still able to use the extension without a hard error.

Reply:
I wanted to trap this in case someone specified `FLASK_STATIC_DIGEST_GZIP_FILES = []` and FLASK_STATIC_COMPRESSION = ["gzip"] or the opposite as you could end up with poor behavior. Perhaps I am being too careful?